### ISSUE TITLE
Provide guidelines for compiling a custom kernel

### ISSUE THREAD

**Issue by luxas** [firecracker-microvm/firecracker#881]
Hi,

Thanks for this great project!
I tried to use the stock Ubuntu 18.04 or 18.10 kernels (4.15 and 4.18), but it stopped working at the point of mounting the filesystem.

I saw the kconfig and will tomorrow try to compile a 4.14 kernel from that config, but I'd be interested in official guidelines how to compile a compliant kernel (any patches or special values). Could you please provide that, and your quickstart kernel's modules? That'd be really helpful, thanks!

**Comment by raduweiss** [comment]
@luxas, indeed, for now all we have is the [kernel config](https://github.com/firecracker-microvm/firecracker/blob/master/resources/microvm-kernel-config) that will work with Firecracker.

There's two things that we need to do here:
- [ ] Provide more clear guidelines on what config settings have what effect (that goes for the  command line parameters.
- [ ] Test newer guest kernels, and figure out the right cadence for pairing new Firecracker releases to new Kernel releases.

My personal take on the latter is that we should probably improve our CI infrastructure to the point where we can always bring in the latest stable Linux kernel release. Then there's also an open question on wether we will only support a single Firecracker release / Linux kernel version, and on how many old previous releases we'll support.

**Comment by acatangiu** [comment]
Link the `resources/microvm-kernel-config` file in the documentation.

Document the minimally required kernel config params for Firecracker to function (check if the default works).

**Comment by alindima** [comment]
We now have a per-feature guest kernel config breakdown: https://github.com/firecracker-microvm/firecracker/blob/main/docs/kernel-policy.md#guest-kernel-modules, as well as a kernel support policy document
